astropy/cosmology/core.py

```diff
@@ -135,11 +136,14 @@ def __init__(self, *args, name=None, meta=None, **kwargs):
         self._name = name
         self.meta.update(meta or {})
 
-    # Make initial signature of cosmology. Overwritten in subclasses.
-    # (Need to make one here b/c Cosmology is not abstract).
-    _init_signature = signature(__init__)
-    _init_signature = _init_signature.replace(
-        parameters=list(_init_signature.parameters.values())[1:])
+    @classproperty
```

Comment:
Looking at this, I'm wondering if this wouldn't make more sense as a regular class property - if you use the `classproperty(lazy=True)`, I think you could delete the copy of this code in `__init_subclass__`...

Reply:
Good suggestion. I'll look into that.

Reply:
Another small advantage may be that `del cls._init_signature` will gain meaning (in case one messes up, as you mentioned in #11817)...